components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java

```diff
@@ -2886,6 +2887,399 @@ private OMElement queryFirstAccess(String columnFamily)
+                statement.setString(index, toDate);
+
+                rs = statement.executeQuery();
+                while (rs.next()) {
+                    int successRequestCount = rs.getInt(APIUsageStatisticsClientConstants.SUCCESS_REQUEST_COUNT);
+                    int throttledOutCount = rs.getInt(APIUsageStatisticsClientConstants.THROTTLED_OUT_COUNT);
+                    int year =  rs.getInt(APIUsageStatisticsClientConstants.YEAR);
+                    int month =  rs.getInt(APIUsageStatisticsClientConstants.MONTH);
+                    String time;
+                    if (APIUsageStatisticsClientConstants.GROUP_BY_HOUR.equals(groupBy)) {
+                        time = rs.getString(APIUsageStatisticsClientConstants.TIME);
+                    } else {
+                        int day =  rs.getInt(APIUsageStatisticsClientConstants.DAY);
+                        time = year + "-" + month + "-" + day + " 00:00:00";
+                    }
+                    throttlingData.add(
+                            new APIThrottlingOverTimeDTO(apiName, provider, successRequestCount, throttledOutCount,
+                                    time)
+                    );
+                }
+
+            } else {
+                throw new APIMgtUsageQueryServiceClientException(
+                        "Statistics Table:" + APIUsageStatisticsClientConstants.API_THROTTLED_OUT_SUMMARY +
+                                " does not exist.");
+            }
+
+            return throttlingData;
+
+        } catch (SQLException e) {
+            throw new APIMgtUsageQueryServiceClientException("Error occurred while querying from JDBC database", e);
+        } finally {
+            if (rs != null) {
+                try {
+                    rs.close();
+                } catch (SQLException e) {
+                    //this is logged and the process is continued because the query has executed
+                    log.error("Error occurred while closing the result set from JDBC database.", e);
+                }
+            }
+            if (statement != null) {
+                try {
+                    statement.close();
+                } catch (SQLException e) {
+                    //this is logged and the process is continued because the query has executed
+                    log.error("Error occurred while closing the prepared statement from JDBC database.", e);
+                }
+            }
+            if (connection != null) {
+                try {
+                    connection.close();
+                } catch (SQLException e) {
+                    //this is logged and the process is continued because the query has executed
+                    log.error("Error occurred while closing the JDBC database connection.", e);
+                }
+            }
+        }
+    }
+
+    /** Given Application name and the provider, returns throttle data for the APIs of the provider invoked by the 
```

Comment:
Incorrect format, you should leave a blank line at the top before the comment. Also finish the line with a full stop.